partiql-parser/src/partiql.pest

```diff
@@ -9,7 +10,7 @@ Query = _{ SOI ~ Token+ ~ EOI}
 // Entry point for query "scanning"
 // Note that this is factored this way to support an iteration style API
 // where we can call back into this rule on subsequent input.
-Scanner = _{ SOI ~ Token }
+Scanner = _{ SOI ~ !COMMENT ~ Token }
```

Comment:
You should not need to have a negative assertion for this, I am guessing this is here because of the `ANY*` rule above.

There is an interesting API question as to whether the scanner should return comments which would help for use cases like syntax highlighting.  For now I think they should be supported as white space (i.e. ignored by the scanner). we can revisit this later.